### DISCUSSION TITLE
This PDF document probably uses a compression technique which is not supported by the free parser shipped with FPDI (mPDF)

### DISCUSSION THREAD

**Discussion by thedeepbhatt** [mpdf/mpdf#1936] [Q&A] [answered]
Hi I have created a file using mPDF which have dynamic data based on my database. Now I want to append another PDF which I have in system. The file which I want to append is having only one image. So normally when I tried to append file which is having text with command `$_new_file = $mpdf->setSourceFile($_attachemnt_name);` is working fine. But when I try to append file with only image it is giving error 

> Type: setasign\Fpdi\PdfParser\CrossReference\CrossReferenceException
>
>Message: This PDF document probably uses a compression technique which is not supported by the free parser shipped with FPDI. (See https://www.setasign.com/fpdi-pdf-parser for more details)

>More details : 
I have php version 7.2.33
and mPDF version "^8.0"

>What I have done so far : 
`$source_file` is orginal file's full path that is my main file and `$attach_pdf_images` list of file which need to get append inside main file.
normal PDF file with text gets appended without any error. PDF File which contains only image are not getting append and giving above error.

this is my code : 
>$mpdf  = new \Mpdf\Mpdf(['mode' => 'utf-8', 'format' => [240, 300]]);
 > // Set source file
        $main_file = $mpdf->setSourceFile($source_file);
        // Loop through all pages of main file
        for ($i = 1; $i <= $main_file; $i++) { 
            $tplId = $mpdf->ImportPage($i); // Import page i
            $mpdf->AddPage(); // Add a new page
            $mpdf->UseTemplate($tplId); // Insert template at default position and size
        }
        >foreach ($attach_pdf_images as $_attachemnt_name) {
            if (file_exists($_attachemnt_name)) {
                if (strtolower(pathinfo($_attachemnt_name, PATHINFO_EXTENSION)) == 'pdf') {
                    $_new_file = '';
                    $mpdf->SetImportUse();
                    $_new_file = $mpdf->setSourceFile($_attachemnt_name);
                    // Loop through all pages of source _new_file
                    for ($i = 1; $i <= $_new_file; $i++) { 
                       $tplId = $mpdf->ImportPage($i); // Import page i
                        $mpdf->AddPage(); // Add a new page
                        $mpdf->UseTemplate($tplId); // Insert template at default position and size
                    }
                }
              }
          }
$mpdf->Output($source_file, 'F');

**Comment by finwe** [comment]
What on the error message

> Message: This PDF document probably uses a compression technique which is not supported by the free parser shipped with FPDI. (See https://www.setasign.com/fpdi-pdf-parser for more details)

is not to understand?

**Reply by thedeepbhatt** [reply]
I understood message but I want solution for it. Please suggest what to do

**Reply by finwe** [reply]
You need to use a source which is compressed with a supported technique. Not much you can do with this stack and your current sources. Furthermore, this is rather to be reported/resolved directly with FPDI authors.

**Reply by thedeepbhatt** [reply]
So what you mean to say is, this is not possible in my case. Only text will get append. If it contain image it will give this error right?

**Reply by finwe** [reply]
No, image can be OK, but only when the source PDF file content is compressed in the way supported by FPDI.

**Reply by thedeepbhatt** [reply]
Ok thanks.

**Reply by thedeepbhatt** [reply]
I found one way should I post it here for some's reference?

**Reply by finwe** [reply]
Well that's up to you, I guess.

**Reply by thedeepbhatt** [reply, marked as answer]
Ok. So found solution for this.
>https://github.com/xthiago/pdf-version-converter

I follow the same process just few things added before appending  PDF. 
Added point to check the version, If version is more then 1.4 then I convert that PDF to 1.4 version. 
To check and convert I use above mention lib. Hope someone need it.
